### DISCUSSION TITLE
The Splink Altair "blocking rules" chart is not displayed correctly when saved as PNG

### DISCUSSION THREAD

**Discussion by aalexandersson** [vega/altair#3336] [Q&A] [answered]
The Splink Altair [blocking rules chart](https://moj-analytical-services.github.io/splink/charts/cumulative_num_comparisons_from_blocking_rules_chart.html) is displayed correctly in Vega Editor, see this [shareable URL](https://vega.github.io/editor/#/url/vega-lite/N4Igxg9gdgZglgcxALlANzgUwO4tJKAFzigFcJSBnAdTgBNCALFAZgAY2AacaYsiygAlMiRoVYcAvpO50AhoTl4QUOQFtMKEPMUBaAKwAWAOwBGQwCYAHADYwcuSwBGATkxOrLm4dNgb+lzorKzBTCxsQGRA1OQAnAGstJziQbkwoSDoSJFQeABsIWOV4TDy6LVjSPM1uaoR08uQyPLzuSntq5XbGTA0te0JMBEKATws2MEiowrpMItySsv7SNSqFODRMAH1YiGxKSO5CCAg84gAHFABtUEXGkErOo7hCTuQQAGUARQAZAAIAMLQLLEaCpECEEbnTTvKAQNQkOR5Q63LBLd67bBbSCkIjgmCFGLid7cZ6vGEgIFqc5xOCUaCUP4AcXScwUmHKRyhFIAjqQ5EQXutNiiQHdlqs8sLtpiDtwCbEiVpSRCXm9KSs1sRNoD4TTYnSGeDIdCtHyBcRFNrNDJUaV7vZYoNKHABfjCQplca1RSAXFna6oH8AAq7OikMCEP4QGB-ACSUHOpCjABEFEouab3ubBVaNjbOHb0Q8ORHQVAdusIN7yVoAEqlyNwaB-OtVv4ACjAmql1r+soA9I6AwK-ucw2WAJTG7laOEI1TIyQAXW4AA9imj7pRFE6a+qqfrDVBGSyoGzBnQ-k4Rq2qpgO5Rp5nefzc9LDiA1xZN-aJVr8x2PYDiiEZf2LR4ahAek92QK4QF0b8QFXVVa3eb5-gAIQKMB4myO9OmkbgelEYlQB3TBLmQcYoktN5QEGNdiUpCgiGjWMAEE6BBZtF11alaXpE9mVZWJ2SvG8-gAUTkMBGD+bCIFw-C63vcFKFIJw6IpDsgVxQhGUwNcwDyUhZj+SABINITGSRWJMDkOhb3qc8xMva9b3HTAMAEft7yfT9sHoJgUEMfQuBAAASbpeiUd4xEIc5KGQAcB02BA5AAOgQF5GE0zLmwHGKYjSoY5F0PIXkwNL9EylxMpYTKACshPBHQ5EoTADOUdqDBMcxrDsBxnDcDwvB8PwAiCEIwgiODQExbFWOJfQrHYUwWG4SCtA7PJMoAHRAKUdy2VQNEOv4AF5+wOo6OsIU71EwQ7Jz+DiADkU07PbDroCAnAu67Yluv6AZAZ9wB7aUgP2FBVvWzbwH9TAXTdZBDEMFwLFMUwbDYFgzEMLbG3LStQS0AAVHp+1OSqoAQfsSd4-t2yCpg-hGChYj+HLNiDJwcLw+m-OqR9XrpP42DqlwZZlzK-ipiX7K8zqiEZJhMEZ8MmxbEg-g1-WIEUPI-jINQnDmdiLL1QSGT+cNNeODmuZFh8n0y9Td2JNhbQePYlv0lAcbYKx8a2tT3l2274FiE6zuekArpuw6Y7jp6Xrez7vpB-7AeT7Rc-B8Fu0laHZRQGwXBYKxce4YcUcDULMex3H8cJ4ntdJtzm0p6ndhafD7M75nu4gP42fkznSG53n0mvQXB-vMW-glqXZZlqx5cVxllfs1WDP1vumd1oMDeOY3TZWC3uZja2rOPRkHcN53p9dsWPbaL24bWtgNt9xacR4mQDYFgFgwHh3VFHQ6AAvOA5w87AxgXAjOH0vpQOgkZBBt1OprhQVndBx0HrxywYdQhj1zrg0zmgn6Yo4CxyIenROQNo50LThQiGJcAKbBhgcZAxh9DGHxlYOuyNUZQCbljHGeMCbmA7mWXiZMe7vCpprfudMGZD3kS2Ue49covxnvmfmC9haQWXqvaWstN4K0YErSie90gHwNponWQY9ZnyNkiS+5tLa30skeGy9tSCOzHlPbmpj3ae39BXKuNcbArmkEAA). The Vega Editor exports the chart correctly. The issue is that the chart saved to PNG using [chart.save('chart.png')](https://altair-viz.github.io/user_guide/saving_charts.html) is not displayed correctly.

I am an end-user. I originally raised the [Splink issue 1990](https://github.com/moj-analytical-services/splink/discussions/1990) against Splink but after learning that the chart is correct in the Vega Editor, I now think the issue instead is with Altair: Altair does not correctly save the Splink "blocking rules" Altair chart as PNG. 

Here is the correct Altair chart before saving:
![blocking_html](https://github.com/altair-viz/altair/assets/20467610/73fb50c9-0ec1-42b2-b5d9-37d3770e584c)

Here is the distorted chart after saving it as PNG. Note how much smaller the saved PNG chart is, possibly because of the unwanted added white space to the left of the y-title:

![block](https://github.com/altair-viz/altair/assets/20467610/485d0177-f7a3-4c7b-841b-cdda2b1b0ef7)

**Comment by mattijn** [comment, marked as answer]
Thank you for raising! I can reproduce the issue as such:
```python
import pandas as pd
import altair as alt

df = pd.DataFrame.from_records([
      {
        "row_count": 583013,
        "rule": "(l.\"last_name\" = r.\"last_name\") AND (l.\"dob\" = r.\"dob\")",
        "cumulative_rows": 583013,
        "cartesian": 449211603714,
        "reduction_ratio": "The rolling reduction ratio with your given blocking rule(s) is 0.999999. This represents the reduction in the total number of comparisons due to your rule(s).",
        "start": 0
      },
      {
        "row_count": 110803,
        "rule": "(l.\"first_name\" = r.\"first_name\") AND (l.\"dob\" = r.\"dob\")",
        "cumulative_rows": 693816,
        "cartesian": 449211603714,
        "reduction_ratio": "The rolling reduction ratio with your given blocking rule(s) is 0.999998. This represents the reduction in the total number of comparisons due to your rule(s).",
        "start": 583013
      },
      {
        "row_count": 63222,
        "rule": "(l.\"zip\" = r.\"zip\") AND (l.\"sex\" = r.\"sex\") AND (l.\"last_name\" = r.\"last_name\") AND (l.\"first_name\" = r.\"first_name\")",
        "cumulative_rows": 757038,
        "cartesian": 449211603714,
        "reduction_ratio": "The rolling reduction ratio with your given blocking rule(s) is 0.999998. This represents the reduction in the total number of comparisons due to your rule(s).",
        "start": 693816
      }
    ]
)

chart = alt.Chart(df).mark_bar().encode(
    x='start',
    x2='cumulative_rows',
    y=alt.Y('rule').sort('-x'),  # .axis(labelLimit=500)
    color=alt.Color('rule').legend(None).scale(scheme='category20c')
)
chart
```
![image](https://github.com/altair-viz/altair/assets/5186265/cbff010a-977c-4493-a5db-3f438d87b834)

```python
import vl_convert as vlc
from IPython.display import Image

png_data = vlc.vegalite_to_png(vl_spec=chart.to_dict())
Image(png_data)
```
![image](https://github.com/altair-viz/altair/assets/5186265/98e08894-50c4-4f98-bdc2-46f893787d43)

As you can wee we are using vl_convert to export the specification into a png-file. Therefore raised the following issue within the vl_convert repository: https://github.com/vega/vl-convert/issues/152
